null_safety_examples/misc/lib/samples/spacecraft.dart

```diff
@@ -0,0 +1,87 @@
+// ignore_for_file: sort_constructors_first
+// #docregion class
+class Spacecraft {
+  String name;
+  DateTime? launchDate;
+
+  // Constructor, with syntactic sugar for assignment to members.
+  Spacecraft(this.name, this.launchDate) {
+    // Initialization code goes here.
+  }
+
+  // Named constructor that forwards to the default one.
+  Spacecraft.unlaunched(String name) : this(name, null);
+
+  int? get launchYear => launchDate?.year; // read-only non-final property
+
+  // Method.
+  void describe() {
+    print('Spacecraft: $name');
+    var launchDate = this.launchDate; // Type promotion doesn't work on getters.
+    if (launchDate != null) {
+      int years = DateTime.now().difference(launchDate).inDays ~/ 365;
+      print('Launched: $launchYear ($years years ago)');
+    } else {
+      print('Unlaunched');
+    }
+  }
+}
+// #enddocregion class
+
+// #docregion extends
+class Orbiter extends Spacecraft {
+  double altitude;
+
+  Orbiter(String name, DateTime launchDate, this.altitude)
+      : super(name, launchDate);
+}
+// #enddocregion extends
+
+// #docregion mixin
+class Piloted {
```

Comment:
Can we use an explicit `mixin` here?

Reply:
That makes sense, added in https://github.com/dart-lang/site-www/pull/2919/commits/32c509f52811be97a5ee91ba8d0bc67130d3ec44